src/bootstrap/src/core/builder.rs

```diff
@@ -2028,10 +2028,16 @@ impl<'a> Builder<'a> {
         // When we build Rust dylibs they're all intended for intermediate
         // usage, so make sure we pass the -Cprefer-dynamic flag instead of
         // linking all deps statically into the dylib.
-        if matches!(mode, Mode::Std | Mode::Rustc) {
+        if matches!(mode, Mode::Std) {
```

Comment:
I think you will need to add `Mode::Codegen` here as `-Cprefer-dynamic` is no longer implied when linking librustc_driver.so

Reply:
I don't see how there's any change to `-C prefer-dynamic` for other crates?